### PR TITLE
663 result page layout download

### LINKED ISSUE #632: Download ballots - add election ID
As Is:
ballot_id,a,b,c,d
a16315ff-068a-46bc-85cb-145529e032f4,0,0,0,0
09f24fdc-c7fa-4b3a-a5cd-40764589d17e,0,0,0,0
9c91811b-b4ba-4fad-9ab1-05ef74920a24,0,0,0,0

Should be:
https://dev.star.vote/x3kbxk/admin/ballots

ballot_id,a,b,c,d
a16315ff-068a-46bc-85cb-145529e032f4,0,0,0,0
09f24fdc-c7fa-4b3a-a5cd-40764589d17e,0,0,0,0
9c91811b-b4ba-4fad-9ab1-05ef74920a24,0,0,0,0

### LINKED ISSUE #663: Results Page Layout + Anonymized Ballot Download
Here's a couple of improvements to make to the results page

- [ ] **Remove race dropdowns**: Currently multi race elections have a separate dropdown for each race. Mark F. , suggested that it would be cleaner if we removed those and I agree. Let's have the races be fully expanded, wither the paper wrapper component removed and separated by ``<hr>`` tags (but keep the existing details dropdowns)
- [ ]  **Calls to action**: At the bottom we should have buttons for "Download Anonymized CSV", "Share Poll/Election" and "Donate" (see thanks page for reference, Share should only be shown on public elections)
- [ ] **Download CSV Setting**: Currently download CSV is only available to admins under the ballot view, but I think it's best practice to have this also shared with voters. This should be a setting that admins can disable, but it should be enabled by default
- [ ] **CSV Bug**: Currently the CSV only displays a single race, but it should include all races. This allows researchers to correlate votes across races. I'm not sure if there's a standard for representing header hierarchies, but "<race name>!!<candidate name>" is a pattern I've seen before

### PR DESCRIPTION
**Pull request by csamuele**
## Description
updated layout of results page and fixed the csv download

## Screenshots / Videos (frontend only) 
localhost_3000_w63bvr_results (2)
image
[Ballot Data - Copy of Copy of US Presidential Election 2024.csv](https://github.com/user-attachments/files/17070628/Ballot.Data.-.Copy.of.Copy.of.US.Presidential.Election.2024.csv)

> Be sure to include both desktop and mobile views (mobile could be as low as 320 px wide) 

## Related Issues

closes #663 
closes #632

### CODE REVIEW

File: packages/frontend/src/components/Election/Results/DowloadCSV.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        let tempCsvData = data.ballots.map(ballot => {
+            let row = {ballot_id: ballot.ballot_id}
+            ballot.votes.forEach(vote => vote.scores.forEach(score => {
+                row[`${vote.race_id}-${score.candidate_id}`] = score.score
+            }));
+            return row
+        })
+        setcsvHeaders(header)
+        setcsvData(tempCsvData)
+        return false
```
Comment (ArendPeter) [inline packages/frontend/src/components/Election/Results/DowloadCSV.tsx:31]: Does this line do anything?

File: packages/frontend/src/components/Election/Results/ViewElectionResults.tsx [hunk 2]
```diff
@@ -3,20 +3,24 @@ import { useParams } from "react-router";
...
+import { useGetBallots } from '~/hooks/useAPI';
 
 const ViewElectionResults = () => {
     
     const { election } = useElection()
     
     const { data, isPending, error, makeRequest: getResults } = useGetResults(election.election_id)
     useEffect(() => { getResults() }, [])
-
+	const { data: ballotsData, isPending: isPendingBallots, error: ballotsError, makeRequest: fetchBallots } = useGetBallots(election.election_id)
```
Comment (ArendPeter) [inline packages/frontend/src/components/Election/Results/ViewElectionResults.tsx:21]: Can you double check the permissions on the useGetBallots api in the backend?

I expect the API is only available to the election admin, but we want that to change based on the public_download setting

**Review by ArendPeter — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/Election/Results/DowloadCSV.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+interface Props {
+	data: {
+		election: Election;
+		ballots: Ballot[];
+	};
+}
+export const DownloadCSV = ({data}: Props) => {
+	const [csvData, setcsvData] = useState([])
+    const [csvHeaders, setcsvHeaders] = useState([])
+	const buildCsvData = () => {
```
Comment (ArendPeter) [inline packages/frontend/src/components/Election/Results/DowloadCSV.tsx:16]: nit: fix tabbing on this file

File: packages/frontend/src/components/Election/Results/ViewElectionResults.tsx (hunk 2 above)
Comment (mikefranze) [reply packages/frontend/src/components/Election/Results/ViewElectionResults.tsx:21]: Do we need an extra setting or should it be based on public_results?

File: packages/frontend/src/components/Election/Results/DowloadCSV.tsx (hunk 1 above)
Comment (mikefranze) [reply packages/frontend/src/components/Election/Results/DowloadCSV.tsx:31]: Looks like it was copied over from my old code. It looks like that old function probably used to return a valid indicator before continuing to something else. Probably fine to remove it.

**Review by ArendPeter — APPROVED** [review]
Amazing work!

File: packages/frontend/src/components/Election/Results/ViewElectionResults.tsx (hunk 2 above)
Comment (ArendPeter) [reply packages/frontend/src/components/Election/Results/ViewElectionResults.tsx]: We elaborated further in slack: https://starvoting.slack.com/archives/C01EBAT283H/p1726845204279549

**Pull request merged**